libs/api/domains/watson-assistant-chat/src/lib/watson-assistant-chat.service.ts

```diff
@@ -0,0 +1,38 @@
+import jwt from 'jsonwebtoken'
+import { ConfigType } from '@island.is/nest/config'
+import { WatsonAssistantChatIdentityTokenInput } from './dto/watsonAssistantChatIdentityToken.input'
+import { WatsonAssistantChatConfig } from './watson-assistant-chat.config'
+import RSA from 'node-rsa'
+
+export class WatsonAssistantChatService {
+  private rsaKey!: RSA
+
+  constructor(private config: ConfigType<typeof WatsonAssistantChatConfig>) {
+    this.rsaKey = new RSA(this.config.directorateOfImmigrationPublicIBMKey)
+  }
+
+  async createIdentityToken(input: WatsonAssistantChatIdentityTokenInput) {
+    const encryptedUserPayload = this.rsaKey.encrypt(
```

Comment:
I'm wondering if this could just be done using NodeJS built in `crypto.publicEncrypt` instead of introducing a npm package. I haven't tried it but from short google it seems to be possible. I would be open to look into that with you if you want.

Reply:
Here's the guide I followed: https://cloud.ibm.com/docs/assistant?topic=assistant-deploy-zendesk

![image](https://user-images.githubusercontent.com/43557895/232464499-4cb39f95-92e3-4461-9378-73760191087a.png)

So this has been tested using the node-rsa library and we know that works.

However we could try using the crypto package instead but how would I encrypt the user payload in the same way?

Reply:
I'll try this out: 
![image](https://user-images.githubusercontent.com/43557895/232738582-b16646a3-8942-47ff-9601-ab679957b621.png)